--- xcsp-parser/AST.hh	2011-03-23 17:57:18.323461739 +0100
+++ xcsp-parser-orig/AST.hh	2008-06-02 18:53:44.000000000 +0200
@@ -615,13 +615,12 @@
     virtual C_AST *makeCTree() const
     {
       C_AST_ListNode *node=reinterpret_cast<C_AST_ListNode *>
-	(malloc(sizeof(C_AST_ListNode)+(list.size()*sizeof(C_AST *))));
+	(malloc(sizeof(C_AST_ListNode)+sizeof(void*[list.size()])));
 
-      node->items=reinterpret_cast<C_AST **>(node+1);
       node->type=LIST;
       node->size=list.size();
 
-      for(unsigned int i=0;i<list.size();++i)
+      for(int i=0;i<list.size();++i)
 	node->items[i]=list[i]->makeCTree();
 
       return reinterpret_cast<C_AST *>(node);
@@ -750,9 +749,8 @@
     virtual C_AST *makeCTree() const
     {
       C_AST_DictNode *node=reinterpret_cast<C_AST_DictNode *>
-	(malloc(sizeof(C_AST_DictNode)+(dict.size()*sizeof(C_AST_DictEntry))));
+	(malloc(sizeof(C_AST_DictNode)+sizeof(C_AST_DictEntry[dict.size()])));
 
-      node->items=reinterpret_cast<C_AST_DictEntry *>(node+1);
       node->type=DICT;
       node->size=dict.size();
 
@@ -942,9 +940,8 @@
     virtual C_AST *makeCTree() const
     {
       C_AST_FxNode *node=reinterpret_cast<C_AST_FxNode *>
-	(malloc(sizeof(C_AST_FxNode)+(nbarg*sizeof(C_AST*))));
+	(malloc(sizeof(C_AST_FxNode)+sizeof(void*[nbarg])));
 
-      node->args = reinterpret_cast<C_AST **>(node+1);
       node->type=type;
       node->nbarg=nbarg;
       for(int i=0;i<nbarg;++i)
@@ -1101,8 +1098,8 @@
     {
       checkargs();
 
-      long prod=1;
-      long x=args[0]->value(varValuation);
+      long long prod=1;
+      long long x=args[0]->value(varValuation);
       int n=args[1]->value(varValuation);
       int test;
 
